### ISSUE TITLE
GDScript 2.0: Function return type `enum` error (Trying to return value of type "int" from a function which the return type is "Dictionary")

### ISSUE THREAD

**Issue by stmSi** [godotengine/godot#57966]
### Godot version

v4.0.alpha2.official [79077e6c1]

### System information

5.13.19-2-MANJARO XFCE, Nvidia 2060

### Issue description

Work fine in alpha 1 (v4.0.alpha1.official  @[31a7ddbf8])

Error exists in alpha 2 (v4.0.alpha2.official [79077e6c1])

Also Auto-Completion does not show the error.

### Steps to reproduce

```
enum State {
	Idle,
	Run,
	Attack,
}


func _ready() -> void:
	getEnum()

func getEnum() -> State:
	return State.Run
```

error: `Trying to return value of type "int" from a function which the return type is "Dictionary".`

**Comment by Mickeon** [comment]
Could it have been caused by https://github.com/godotengine/godot/pull/57591 ?

**Comment by stmSi** [comment]
> Could it have been caused by #57591 ?

Yes. I saw multiple comments there stating this error issue.

**Comment by jmb462** [comment]
I confirm that it is caused by ceafdf347e5ecc050629fd4eac93030dabb0d0e9

**Comment by jmb462** [comment]
I confirm that this issue still occurs in current master branch. (and in beta 10)

**Comment by adamscott** [comment]
CC @anvilfolk

**Comment by anvilfolk** [comment]
Also fixed by #69590 :)

Thanks @adamscott!
